src/ScoreTrack.py

```diff
@@ -16,42 +18,144 @@
+#         claim_game_counter = int(claim[0])
+#         t=claim[1:]
+#         for score_and_init in t:
+#             #print("record= ",claim_game_counter,score_and_init)
+
+#             for recent_game_index, recent_game in enumerate(recent_scores):
+#                 #print("recent game ",recent_game)
+#                 recent_game_counter = int(recent_game[0])
+#                 recent_scores_and_initials = recent_game[1:]
+#                 #print("c=",recent_game_counter,"i=",recent_scores_and_initials)
+
+#                 if (claim_game_counter == recent_game_counter):
+#                     #print("match")
+#                     # Check each score
+#                     for score_index, i in enumerate(recent_scores_and_initials):
+#                         #print("i=",i,i[1],score_and_init[1])
+#                         if int(i[1]) == int(score_and_init[1]):   #score match
+#                             if not i[0]:
+#                                 print ("SCORE: new score:",score_and_init,recent_game_index,score_index)
+#                                 recent_scores[recent_game_index][1 + score_index] = (
+#                                     score_and_init[0].upper(),  # New initials
+#                                     i[1],  # Existing score
+#                                 )
+
+#                                 year, month, day, _, _, _, _, _ = rtc.datetime()
+#                                 new_score = {
+#                                     'initials': score_and_init[0].upper(),
+#                                     'full_name': '    ',
+#                                     'score':  int(i[1]),
+#                                     'date': f"{month:02d}/{day:02d}/{year}",
+#                                     "game_count": SharedState.gameCounter
+#                                 }
+#                                 update_leaderboard(new_score)
+#     return "ok"
+
+
+def claim_score(initials, score):
+    # claim a score
+    global recent_scores
+
+    initials = initials.upper()
+
+    for game_index, game in enumerate(recent_scores):
+        for player_index, player in enumerate(game[1:], start=1):
+            if player[0] == "" and player[1] == score:
+                print("SCORE: new score:", initials, score, game_index, player_index)
+                recent_scores[game_index][player_index] = (initials, score)
+                year, month, day, _, _, _, _, _ = rtc.datetime()
+                new_score = {
+                    "initials": initials,
+                    "full_name": None,
+                    "score": score,
+                    "date": f"{month:02d}/{day:02d}/{year}"
+                    # TODO game count does not appear to be used
+                    # "game_count": SharedState.gameCounter
+                }
+                update_leaderboard(new_score)
+                return
+
+    raise ValueError("SCORE: Score not found in claim list")
```

Comment:
I simplified this function a lot, it seems like most of that logic is trying to figure out which game to claim the score from, but in the end I don't think it matters. if the player number and score is right, it should be fine. 

Also I redesigned a little so it's one call per claimed score. The Java script can handle that more easily and it make the code easier to follow.